--- old/src/CherryPy-3.2.2/cherrypy/_cpwsgi.py.orig	2016-05-31 13:31:20.017859193 +0100
+++ new/src/CherryPy-3.2.2/cherrypy/_cpwsgi.py	2016-05-31 13:32:34.036193226 +0100
@@ -268,8 +268,9 @@ class AppResponse(object):
         """Create a Request object using environ."""
         env = self.environ.get
         
-        local = httputil.Host('', int(env('SERVER_PORT', 80)),
-                           env('SERVER_NAME', ''))
+        local = httputil.Host('',
+                              int(env('SERVER_PORT', 80) or -1),
+                              env('SERVER_NAME', ''))
         remote = httputil.Host(env('REMOTE_ADDR', ''),
                                int(env('REMOTE_PORT', -1) or -1),
                                env('REMOTE_HOST', ''))
